src/MyHealth/SymptomLogContext.tsx

```diff
@@ -84,31 +89,56 @@ export const SymptomLogProvider: FunctionComponent = ({ children }) => {
     setDailyLogData(combineSymptomAndCheckInLogs(logEntries, checkIns))
   }, [checkIns, logEntries])
 
+  const failureResponse = (errorMessage: string): OperationResponse => {
+    Logger.error(errorMessage)
+    return { kind: "failure" }
+  }
+
   const addLogEntry = async (symptoms: Symptom[]) => {
-    const newEntry = {
-      symptoms,
-      date: Date.now(),
+    try {
+      const newEntry = {
+        symptoms,
+        date: Date.now(),
+      }
+      await createLogEntry(newEntry)
+      await fetchLogEntries()
+      return SUCCESS_RESPONSE
+    } catch (e) {
+      return failureResponse(e.message)
     }
-    await createLogEntry(newEntry)
-    await fetchLogEntries()
   }
 
   const updateLogEntry = async (updatedLogEntry: SymptomLogEntry) => {
-    await modifyLogEntry(updatedLogEntry)
-    await fetchLogEntries()
+    try {
+      await modifyLogEntry(updatedLogEntry)
+      await fetchLogEntries()
+      return SUCCESS_RESPONSE
+    } catch (e) {
+      return failureResponse(e.message)
```

Comment:
What do you think about added `Logger.error()`'s for these operations?  Maybe it makes sense to do this in the native module?

Reply:
I believe we are logging the errors on the `failureResponse` method. Would it be better to move it higher to the `nativeModule`?

Reply:
ah, no, just misread the code. imo, its fine where it is at.